ruby/highlight_search_result/lib/highlight_search_result.rb

```diff
@@ -0,0 +1,9 @@
+class StringModifier
+  def search_element(search_element, incoming_string)
+    incoming_string.gsub(%r(#{search_element})i, '(' + search_element + ')')
```

Comment:
Method `gsub` is for substitution not for searching. Read rubydoc for String class.